src/computation.cpp

```diff
@@ -88,42 +92,181 @@ void Computation::runSimulation()
+    {
+        // bottom
+        discretization_->u(i, discretization_->uJBegin() +1) = settings_.dirichletBcBottom[1];
+        
+        // top
+        discretization_->u(i, discretization_->uJEnd() -1) = settings_.dirichletBcTop[1];
+    }
+
+    // sides - they take priority then
+    // set u
+    for ( int j = discretization_->uJBegin(); j < discretization_->uJEnd(); j++)
+    {
+        // left
+        discretization_->u(discretization_->uIBegin() +1,j) = settings_.dirichletBcLeft[0];
+
+        // top
+        discretization_->u(discretization_->uIEnd() -1,j) = settings_.dirichletBcRight[0];
+    }
+    // set v
+    for ( int j = discretization_->vJBegin(); j < discretization_->vJEnd(); j++)
+    {
+        // left
+        discretization_->v(discretization_->vIBegin() +1,j) = 2 * settings_.dirichletBcLeft[1] - discretization_->v(discretization_->vIBegin() +2, j);
+
+        // top
+        discretization_->v(discretization_->vIEnd() -1,j) = 2 * settings_.dirichletBcRight[1] - discretization_->u(discretization_->vIEnd() -2, j);
+    }
+
+    // set p
+    pressureSolver_.setBoundaryValues()
+
+    // set f,g boundary conditions the same as u,v
+    // bottom and top
+    // set f
+    for ( int i = discretization_->uIBegin(); i < discretization_->uIEnd(); i++)
+    {
+        // bottom
+        discretization_->f(i, discretization_->uJBegin() +1) = discretization_->u(i, discretization_->uJBegin() +1);
+        
+        // top
+        discretization_->f(i, discretization_->uJEnd() -1) = discretization_->u(i, discretization_->uJEnd() -1);
+    }
+    // set g
+    for ( int i = discretization_->vIBegin(); i < discretization_->vIEnd(); i++)
+    {
+        // bottom
+        discretization_->g(i, discretization_->uJBegin() +1) = discretization_->u(i, discretization_->uJBegin() +1);
+        
+        // top
+        discretization_->g(i, discretization_->uJEnd() -1) = discretization_->u(i, discretization_->uJEnd() -1);
+    }
+
+    // sides - they take priority then
+    // set f
+    for ( int j = discretization_->uJBegin(); j < discretization_->uJEnd(); j++)
+    {
+        // left
+        discretization_->f(discretization_->uIBegin() +1,j) = discretization_->u(discretization_->uIBegin() +1,j);
+
+        // top
+        discretization_->f(discretization_->uIEnd() -1,j) = discretization_->u(discretization_->uIEnd() -1,j);
+    }
+    // set g
+    for ( int j = discretization_->vJBegin(); j < discretization_->vJEnd(); j++)
+    {
+        // left
+        discretization_->g(discretization_->vIBegin() +1,j) = discretization_->v(discretization_->vIBegin() +1,j);
+
+        // top
+        discretization_->g(discretization_->vIEnd() -1,j) = discretization_->v(discretization_->vIEnd() -1,j);
+    }
+
 }
     
 void Computation::computePreliminaryVelocities()
 {
-    return;
+    // calculate F
+    for ( int j = discretization_->uJBegin() +1; j < discretization_->uJEnd() -1; j++)
```

Comment:
warum addierst du hier immer noch 1 auf uJBegin oder uIBegin drauf? ist nicht in uJBegin schon der offset mit eingerechnet?